pilot/pkg/serviceregistry/kube/controller/ambient/services.go

```diff
@@ -44,6 +45,11 @@ func (a *index) ServicesCollection(
 			}
 		}
 		waypoint := fetchWaypoint(ctx, Waypoints, Namespaces, s.ObjectMeta)
+		if waypoint != nil && !waypointForService(waypoint) {
```

Comment:
can we merge it with fecthWaypoint? 

So we can simplify by separating  `fetchWaypointForService` and `fetchWaypointForWorkload`

Reply:
That's a great idea! Updated